rust/agents/relayer/src/msg/pending_operation.rs

```diff
@@ -0,0 +1,175 @@
+use std::cmp::Ordering;
+use std::time::Instant;
+
+use async_trait::async_trait;
+use enum_dispatch::enum_dispatch;
+use eyre::Report;
+
+#[allow(unused_imports)] // required for enum_dispatch
+use super::pending_message::PendingMessage;
+
+/// A pending operation that will be run by the submitter and cause a
+/// transaction to be sent.
+#[async_trait]
+#[enum_dispatch]
+pub trait PendingOperation {
+    /// Prepare to run this operation. This will be called before every run and
+    /// will usually have a very short gap between it and the run call.
+    async fn prepare(&mut self) -> TxPrepareResult {
```

Comment:
nit: Consider renaming the return types for consistency. Some suggestions:

PreparationResult/SubmissionResult/ValidationResult
PrepareResult/SubmitResult/ValidateResult

Reply:
Yeah, that makes sense